frontend/packages/dev-console/src/components/health-checks/AddHealthCheck.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikProps, FormikValues } from 'formik';
+import { Form } from '@patternfly/react-core';
+import { FirehoseResult, ContainerDropdown, history } from '@console/internal/components/utils';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { FormFooter } from '@console/shared';
+import { getResourcesType } from '../edit-application/edit-application-utils';
+import HealthChecks from './HealthChecks';
+
+type AddHelathProps = {
+  resource?: FirehoseResult<K8sResourceKind>;
+  currentContainer: string;
+};
+
+const AddHealthCheck: React.FC<FormikProps<FormikValues> & AddHelathProps> = ({
+  resource,
+  currentContainer,
+  handleSubmit,
+  handleReset,
+  dirty,
+  errors,
+  status,
+  isSubmitting,
+  setFieldValue,
+}) => {
+  const { data } = resource;
+  const [currentKey, setCurrentKey] = React.useState(currentContainer);
+
+  const handleSelectContainer = (containerName: string) => {
+    setCurrentKey(containerName);
+    setFieldValue('containerName', containerName);
+    history.push(
+      `/k8s/ns/${data.metadata.namespace}/${data.kind}/${data.metadata.name}/containers/${containerName}/health-checks`,
+    );
+  };
+
+  const containers = resource.loaded && _.keyBy(data.spec.template.spec.containers, 'name');
+
+  return (
+    <div className="co-m-pane__body">
+      <Form onSubmit={handleSubmit}>
+        <div>
+          Managing health check for:{'  '}
+          <ContainerDropdown
+            currentKey={currentKey}
+            containers={containers}
+            onChange={handleSelectContainer}
+          />
```

Comment:
Missing link back to workload and learn more:
![image](https://user-images.githubusercontent.com/14068621/79658149-5977a900-8181-11ea-984a-1687968ef358.png)
Also notice that for a single container there is no dropdown. Only when there are multiple containers should there be a dropdown:
![image](https://user-images.githubusercontent.com/14068621/79658536-757b4a80-8181-11ea-9a97-0e67d111d560.png)